internal/move2kubeapi/handlers/workspaces.go

```diff
@@ -0,0 +1,261 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"github.com/konveyor/move2kube-api/internal/authserver"
+	"github.com/konveyor/move2kube-api/internal/cloudevents"
+	"github.com/konveyor/move2kube-api/internal/common"
+	"github.com/konveyor/move2kube-api/internal/sessions"
+	"github.com/konveyor/move2kube-api/internal/types"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	workspaceIdRouteVar     = "work-id"
+	projectIdRouteVar       = "proj-id"
+	projectInputIdRouteVar  = "input-id"
+	projectOutputIdRouteVar = "output-id"
+)
+
+func HandleListWorkspaces(w http.ResponseWriter, r *http.Request) {
+	logrus.Debug("HandleListWorkspaces start")
+	var workspaces []types.Workspace
+	if common.Config.AuthEnabled {
+		accessToken, err := common.GetAccesTokenFromAuthzHeader(r)
+		if err != nil {
+			logrus.Errorf("failed to get the access token from the authorization header. Error: %q", err)
+			sendErrorJSON(w, "failed to get the access token from the authorization header", http.StatusBadRequest)
+			return
+		}
+		workspaceIds, err := m2kFS.ListWorkspaceIds()
+		if err != nil {
+			logrus.Errorf("failed to list the workspace ids. Error: %q", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		filteredIds, err := authserver.FilterWorkspacesUserHasAccessTo(workspaceIds, accessToken)
+		if err != nil {
+			if _, ok := err.(types.ErrorTokenExpired); ok {
+				logrus.Errorf("the user access token expired")
+				w.Header().Set(common.AUTHENTICATE_HEADER, common.AUTHENTICATE_HEADER_MSG)
+				w.WriteHeader(http.StatusUnauthorized)
+				return
+			}
+			logrus.Errorf("failed to filter the workspace ids the user has access to. Error: %q", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		workspaces, err = m2kFS.ListWorkspaces(filteredIds)
+		if err != nil {
+			logrus.Errorf("failed to list the workspaces. Error: %q", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+	} else {
+		var err error
+		workspaces, err = m2kFS.ListWorkspaces(nil)
+		if err != nil {
+			logrus.Errorf("failed to list the workspaces. Error: %q", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+	}
+	w.Header().Set(common.CONTENT_TYPE_HEADER, common.CONTENT_TYPE_JSON)
+	w.WriteHeader(http.StatusOK)
+	if err := json.NewEncoder(w).Encode(workspaces); err != nil {
+		logrus.Errorf("failed to send the workspaces json in the response. Error: %q", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	logrus.Debug("HandleListWorkspaces end")
+}
+
+func HandleCreateWorkspace(w http.ResponseWriter, r *http.Request) {
+	logrus.Debug("HandleCreateWorkspace start")
```

Comment:
Do we want to make all these messages as trace?

Reply:
done